### ISSUE TITLE
SelectBox: NVDA does not read the state of select box as "expanded" when it opens

### ISSUE THREAD

**Issue by AccessibilityTestingTeam-TCS** [Microsoft/vscode#52297]
**Environment Details:** 
VSCode Version : 1.24.0
OS Version : Win10

**Additional Details:**
MAS Violated: MAS1.3.1
ScreenReader Version: NVDA2018.2

**Repro Steps:**
1. Launch VS Code while NVDA is ON.
2. Go to View menu and Select Integrated Terminal.
3. Tab till combo box next to Terminal tab item.

**Actual:**
NVDA does not read the state of tree view "expanded" when combobox next to Terminal tab item opens. 

**Expected:**
NVDA should read read its state whenever it is expanded or collapsed.

**Recommendations:**
Refer below link which is repository of bug fixes code snippets:
https://microsoft.sharepoint.com/teams/msenable/mas/pages/browse-fixes.aspx

**User Impact:**
User will not be able to know the control's state.

**MAS Reference**
https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}

Attachment for Reference: 
[A11y VSCode Terminal NVDA not reading state.pptx](https://github.com/Microsoft/vscode/files/2114676/A11y.VSCode.Terminal.NVDA.not.reading.state.pptx)

Does this issue occur when all extensions are disabled?: Yes

**Comment by Tyriar** [comment]
FYI @claudioc

**Comment by AccessibilityTestingTeam-TCS** [comment]
Please refer to the video attached in "Attachment for Reference" section for more clarification.

**Comment by isidorn** [comment]
This is nothing related to the terminal, it is an issue with our windows select box imlementation.
Forwarding to @cleidigh and @bpasero

**Comment by cleidigh** [comment]
I started looking at this, I was not familiar with aria-expanded, I think toggling this class on select box open
for the parents select should be the answer.

When looking at this my NVDA does not seem to read out the options anymore which was added long time ago:

		data.root.setAttribute('aria-label', nls.localize('selectAriaOption', "{0}", optionText));

Do others here are the options ?

**Comment by bpasero** [comment]
@cleidigh seems to work for me:

![image](https://user-images.githubusercontent.com/900690/42369144-5a5defd4-8109-11e8-9093-8a82477f6053.png)

**Comment by cleidigh** [comment]
@bpasero 
okay that's good, maybe I have a configuration issue.

What is the screenshot above?

**Comment by bpasero** [comment]
@cleidigh it is the "Speech Viewer" from NVDA (the one that just shows what is being read instead of reading it)

**Comment by cleidigh** [comment]
I'll try that, it doesn't work very well to have NVDA babbling and Dragon attempting to respond, silent operation will make it easier!

**Comment by cleidigh** [comment]
@AccessibilityTestingTeam-TCS 
@bpasero 


I got myself more familiar with various aria attributes and did some experimentation.
Additionally I looked at other widgets like quickopen.

The selectBox on Windows and Linux is a combination of a native HTML select in the header
and a list widget within a contextView container.  Since the select does not actually
expand (the contextView drop-down list is revealed), we can't rely on the screen reader
doing auto readout of the select.  The list (tree ) gets focus and we can apply
the aria-expanded to that.  Below screenshot shows the reader output for opening the debug
configurations drop-down and changing the selection.

![image](https://user-images.githubusercontent.com/25272315/42424371-329f9066-82d9-11e8-9e7a-d2491d881678.png)

**Comment by bpasero** [comment]
> 1. Add selectBox.ariaLabel() so that each client can set the label to be used
> for the parent select as well as the drop-down list/tree.

Let's discuss in https://github.com/Microsoft/vscode/issues/53821

> 2. Set aria-expanded on the drop-down list

I wonder if that helps though because of the context view being special. I would check with the https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html example

> 3. Pass ariaLabel to the list options

I am not sure that is needed, I would use the option label as aria label, what else would be needed?

> 4. Add aria-selected to the focus list item (exists in tree widget but not list)

Yes, if the tree is doing something that the list is not yet doing, we should align.
